pages/api/users/popular.js

```diff
@@ -7,7 +7,7 @@ import Profile from "../../../models/Profile";
 export default async function handler(req, res) {
   await connectMongo();
 
-  const getProfiles = await Profile.find({}).sort({ views: -1 }).limit(10);
+  const getProfiles = await Profile.find({}).sort({ views: -1 });
```

Comment:
I think this will bring back too many results, currently about 600 and in the future even more.

Reply:
> Btw you don't need to close an open PR to add more changes, you can push new commits to the same branch in your fork and GitHub will update the open PR

Ooo okay, didn't know about it earlier, will make sure to do that next time !

Reply:
> I think this will bring back too many results, currently about 600 and in the future even more.

so maybe making it like .limit(100) and than it will check from the first 100 and sort them out as per the stats are public or not will work ?

Reply:
but still if more than 90 people in the top 100 has their stats hidden than it won't be good ig

Reply:
We dot know if it will work or not because it depends which are the popular profiles and if their stats are public - this can always change also there is no way of knowing

I shared my thoughts in the main part of the PR

> This is a challenging problem to solve. I think it might need paginated recursion to achieve this, but I am open to other ideas

Reply:
> I think binging back all the data is too much, it will load many files.
> 
> This is a challenging problem to solve. I think it might need paginated recursion to achieve this, but I am open to other ideas

But in the worst case, we have to fetch all profiles in that approach as well.